presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparator.java

```diff
@@ -13,88 +13,108 @@
  */
 package com.facebook.presto.tests.statistics;
 
+import com.facebook.presto.Session;
 import com.facebook.presto.cost.PlanNodeStatsEstimate;
-import com.facebook.presto.execution.StageInfo;
+import com.facebook.presto.cost.StatsCalculator;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.type.Type;
 import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.plan.OutputNode;
 import com.facebook.presto.sql.planner.plan.PlanNode;
-import com.facebook.presto.sql.planner.plan.PlanNodeId;
-import com.facebook.presto.sql.planner.planPrinter.PlanNodeStats;
-import com.facebook.presto.sql.planner.planPrinter.PlanNodeStatsSummarizer;
+import com.facebook.presto.testing.MaterializedRow;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
-import java.util.function.BinaryOperator;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import java.util.OptionalDouble;
 
-import static com.facebook.presto.execution.StageInfo.getAllStages;
-import static com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
-import static com.facebook.presto.util.MoreMaps.mergeMaps;
-import static com.google.common.collect.Maps.transformValues;
+import static com.facebook.presto.sql.planner.iterative.Lookup.noLookup;
+import static com.facebook.presto.transaction.TransactionBuilder.transaction;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
 
 public final class MetricComparator
```

Comment:
I don't like the name of these classes because `Metric` is extremely vague and `Comparator` is confusing because this doesn't implement `Comparator`, so it makes it hard to get a sense of what this is for.  I'm trying to think of something better but haven't come up with anything great yet.   Because actually reviewing this now, the code is really nice and straight forward.  This class is about collecting estimated and actual values for the different stats fields for evaluation by MetricCheck. (Maybe instead of `MetricCheck` it could be called `StatsCorrectnessCheck` or `StatsAccuracyCheck`).

Reply:
Hm.. it's just a helper class for `StatisticsAssertion`. I can rename it, but `*Check` is not good either (this class doesn't check anything). `MetricAcqusition`? For now i just made it package private.